rand/rand.go

```diff
@@ -0,0 +1,194 @@
+package rand
+
+import (
+	"fmt"
+	"github.com/wego/pkg/errors"
+	"math"
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"time"
+	"unsafe"
+)
+
+// options to generate random string
+const (
+	Numbers = 1 << iota
+	Letters = 1 << iota
+	Upper   = 1 << iota
+	Lower   = 1 << iota
+
+	numbers                      = "0123456789"
+	letters                      = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	lowerLetters                 = "abcdefghijklmnopqrstuvwxyz"
+	upperLetters                 = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	numbersAndUpperLetters       = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	numberAndLowerLetters        = "0123456789abcdefghijklmnopqrstuvwxyz"
+	numbersAndLetters            = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	numbersLength                = 10
+	lettersLength                = 26
```

Comment:
This should be 52.
Can we use `len(letters)` to make sure all the length are correct?

Reply:
Or we use `len(stringConstantHere)` directly in the `optionMappingLen`?
The `len()` to a string constant is also a constant